myhoard/restore_coordinator.py

```diff
@@ -1128,13 +1137,38 @@ def _rename_prefetched_binlogs_back(self, binlogs):
                 last_renamed_index = remote_index - 1
         self.update_state(last_flushed_index=last_renamed_index, last_renamed_index=last_renamed_index)
 
+    def _sql_thread_not_running(self, slave_status):
+        return slave_status["Slave_SQL_Running"] != "Yes"
+
+    def _start_sql_thread(self, cursor):
+        # Try to start SQL thread with an incremental delay
+        if self.sql_thread_restart_count > 0:
+            factor = 1 if self.sql_thread_restart_count == 1 else (self.sql_thread_restart_count - 1) * 1.5
+            iteration_delay = min(self.SQL_THREAD_RESTART_INIT_INTERVAL * factor, self.SQL_THREAD_RESTART_MAX_INTERVAL)
+            if self.sql_thread_restart_time is not None:
+                time_passed = time.monotonic() - self.sql_thread_restart_time
+                if time_passed < iteration_delay:
+                    return
+        self.sql_thread_restart_time = time.monotonic()
```

Comment:
This should log that it does the restart

Reply:
Added some log.warning about that.